File: src/Mayonaise.cpp
#include "../include/Mayonaise.hpp"

LinkedList<FarmProductType> Mayonaise::resep;

//Konstruktor Cheese
Mayonaise::Mayonaise(int _price,  ProductType _productType, ObjectType _objectType, char _objectSymbol)  {
    initializeProduct(_price, _productType, _objectType, _objectSymbol);
}

void Mayonaise::InitResep() {
    //Inisialisasi resep:
    Mayonaise::getResep().add(EGG);
    Mayonaise::getResep().add(MILK);
}

FarmProductType Mayonaise::getResepwithIdx(int i) {
    return Mayonaise::getResep().get(i);
}

File: Java/src/main/java/engisubs/gameobject/product/side_product/Mayonaise.java
package main.java.engisubs.gameobject.product.side_product;

import java.util.LinkedList;

public class Mayonaise extends SideProduct{
    private static LinkedList<FarmProductType> resep;

    public Mayonaise(int _price){
        super.initializeProduct(_price, ProductType.SIDEPRODUCT, GameObjectType.MAYONAISE, "Nice Sauce for a Nice Food~~");
    }

    public static LinkedList<FarmProductType> getResep(){
        return resep;
    }

    public static FarmProductType getResepwithIdx(int index){
        return getResep().get(index);
    }

    public static void InitResep(){
        resep.add(FarmProductType.EGG);
        resep.add(FarmProductType.MILK);
    }
}